src/main/java/apoc/uuid/UuidHandler.java

```diff
@@ -0,0 +1,205 @@
+    public void afterCommit(TransactionData data, Object state) {
+    }
+
+    @Override
+    public void afterRollback(TransactionData data, Object state) {
+    }
+
+    private void executeUuid(TransactionData txData, String phase) {
+        if (uuid.containsKey("")) updateUuid(null,null);
+        GraphDatabaseService db = properties.getGraphDatabase();
+        Map<String, String> exceptions = new LinkedHashMap<>();
+        Iterable<Node> createdNodes = txData.createdNodes();
+        Iterable<PropertyEntry<Node>> assignedNodeProperties = txData.assignedNodeProperties();
+
+        uuid.forEach((key, data) -> {
+            if (data.get("paused").equals(false)) {
+                try (Transaction tx = db.beginTx()) {
+                    List<String> labels = (List<String>) data.get("labels");
+                    labels.forEach(label -> {
+                        Map<String, Object> params = (Map<String, Object>) data.get("params");
+                        if (when(params, phase)) {
+                            if (createdNodes.iterator().hasNext()) {
+                                createdNodes.forEach(node -> {
+                                    boolean sameLabel = isSameLabel(label, node.getLabels());
+                                    if (sameLabel && !node.hasProperty(UUID_PROPERTY)) {
+                                        String uuid = UUID.randomUUID().toString();
+                                        node.setProperty(UUID_PROPERTY, uuid);
+                                    }
+                                });
+                            }
+                            if (assignedNodeProperties.iterator().hasNext()) {
+                                assignedNodeProperties.forEach(nodePropertyEntry -> {
+                                    boolean sameLabel = isSameLabel(label, nodePropertyEntry.entity().getLabels());
+                                    if (sameLabel && nodePropertyEntry.key().equals(UUID_PROPERTY) && (nodePropertyEntry.value().equals("") || nodePropertyEntry.value() == null)) {
+                                        nodePropertyEntry.entity().setProperty(UUID_PROPERTY, nodePropertyEntry.previouslyCommitedValue());
+                                    }
+
+                                });
+                            }
+                        }
+                    });
+                    tx.success();
+                } catch (Exception e) {
+                    log.warn("Error executing uuid " + key + " in phase " + phase, e);
+                    exceptions.put(key, e.getMessage());
+                }
+            }
+        });
+    }
+
+    private boolean isSameLabel(String label, Iterable<Label> labelIterable) {
+        return StreamSupport.stream(labelIterable.spliterator(), false).allMatch(name -> name.name().equals(label));
+    }
+
+    private boolean when(Map<String, Object> selector, String phase) {
+        if (selector == null) return (phase.equals("before"));
```

Comment:
se anche `phase` è null?
```suggestion
        if (selector == null) return phase.equals("before");
```

Reply:
in questo caso non potrà arrivare mai a null perchè prende il valore del phase che è solo `before` in questo caso

Reply:
non è sufficiente fare solo `if (phase.equals("before"))`?